agent/run.sh

```diff
@@ -100,7 +100,7 @@ launch_agent()
         -e CATTLE_URL \
         -e CATTLE_HOST_LABELS \
         -e CATTLE_VOLMGR_ENABLED \
```

Comment:
I think you need to add `-e HOST_DOCKER_SOCK` here also.  This is important for upgrading the agent.  You can test this by starting rancher/server with this agent, then change /v1/settings/bootstrap.required.image to a new image tag/label.  Then restart rancher-agent.  After the upgrade to the new image see if the container was created properly.